js/saveState.js

```diff
@@ -0,0 +1,42 @@
+import _ from "lodash";
+
+import { COOKIE_LAST_STATE, COOKIE_OMIT_FROM_STATE } from "./constants/AppConstants";
+
+// store the cookie here in case we're testing
+let lastCookie = "";
+
+function cookiesToMap (cookieStr) {
+  return _.chain(cookieStr.split(/;\s*/))
+          .map((keyValStr) => keyValStr.split(/=/))
+          .fromPairs()
+          .value();
+}
+
+export function restoreState() {
+	let cookie = lastCookie;
+	if (typeof document !== 'undefined') {
+    cookie = document.cookie;
+	}
+  const stateEncoded = cookiesToMap(cookie)[COOKIE_LAST_STATE];
```

Comment:
I guess that would help clarify the function's semantics. Currently it's implicit that an empty/missing cookie results in `{}` and making it explicit helps readability.